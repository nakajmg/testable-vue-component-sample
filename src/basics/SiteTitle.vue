<template>
  <div class="SiteTitle">
    {{title}}
  </div>
</template>

<script>
export default {
  name: "SiteTitle",
  props: {
    title: String,
  },
}
</script>


<style lang="scss">
.SiteTitle {
}
</style>
