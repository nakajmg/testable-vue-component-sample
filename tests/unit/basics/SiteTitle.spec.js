import { shallowMount } from "@vue/test-utils"
import SiteTitle from "@/basics/SiteTitle.vue"
describe("SiteTitle.vue", () => {
  const propsData = {
    title: "test title",
  }
  it("props", () => {
    const wrapper = shallowMount(SiteTitle, { propsData })
    expect(wrapper.vm.$props).toEqual(propsData)
  })
  describe("template", () => {
    it("snapshot", () => {
      const wrapper = shallowMount(SiteTitle, { propsData })
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
