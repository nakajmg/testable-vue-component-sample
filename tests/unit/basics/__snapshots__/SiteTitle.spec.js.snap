// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteTitle.vue template snapshot 1`] = `
<div
  class="SiteTitle"
>
  
  test title

</div>
`;
